Dockerfile

```diff
@@ -0,0 +1,179 @@
+ARG NODE_VERSION=11.6.0
+ARG COMPOSER_VERSION=1.8.0
+ARG PHP_VERSION=7.2.13
+ARG ICU_VERSION=63.1
+ARG APCU_VERSION=5.1.16
+ARG XDEBUG_VERSION=2.6.1
+
+
+#####################################
+##               APP               ##
+#####################################
+FROM php:${PHP_VERSION}-fpm as app
+
+ARG ICU_VERSION
+ARG APCU_VERSION
+
+WORKDIR /app
+
+EXPOSE 80
+
+# Install paquet requirements
+RUN export PHP_CPPFLAGS="${PHP_CPPFLAGS} -std=c++11"; \
+    set -ex; \
+    # Install required system packages
+    apt-get update; \
+    apt-get install -qy --no-install-recommends \
+            libzip-dev \
+    ; \
+    # Compile ICU (required by intl php extension)
+    curl -L -o /tmp/icu.tar.gz http://download.icu-project.org/files/icu4c/${ICU_VERSION}/icu4c-$(echo ${ICU_VERSION} | sed s/\\./_/g)-src.tgz; \
+    tar -zxf /tmp/icu.tar.gz -C /tmp; \
+    cd /tmp/icu/source; \
+    ./configure --prefix=/usr/local; \
+    make clean; \
+    make; \
+    make install; \
+    #Install the PHP extensions
```

Comment:
I'd split it as separate `RUN` step, any mistake in app-specific set of `docker-php-ext` requires full rebuild of this later which takes ages.

Reply:
I've add all of it in a single layout to follow Dockerfile best practices that recommend to do the less number of layer as possible. Then, when you want to add something in your image, you can split it to test before, then regroup it after ?

I think, it's not totaly usefull to split this part, because we just need the split sometimes when you want to change something, but otherwise we never change anything.

That's my opinion. But I understand what you say, because when I edit this part I split it for try, because this part (with ICU) is very long.

Wait coments, but if more people prefer, we can split after the ICU install.